### PR TITLE
Feat: add test for Betting and Outcomes

### LINKED ISSUE #123: [Feature]Betting and Outcome Processing Tests
Task: Create tests for betting, flipping coins, and processing payouts.

Criteria:

- Implement tests for place_bet(), play(), and flip_coin().

- Validate token transfers occur correctly when bets are placed.

- Confirm that flip outcomes are randomized and fairly distributed.

- Ensure payouts are correctly calculated based on house edge settings.

Edge cases: Test invalid bets (below min or above max), attempts to flip without placing a bet, and repeated betting without resolving prior flips.

### PR DESCRIPTION
**Pull request by emarc99**
## Description

WIP: This a draft and work in progress

## Related Issues
Closes #123 

## Changes Made
- [ ] List major changes
- [ ] Highlight important updates
- [ ] Describe any breaking changes (if applicable)
- [ ] New and existing unit tests pass locally with my changes

## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] Documentation has been updated accordingly

### CODE REVIEW

File: packages/snfoundry/contracts/Scarb.lock
Comment (jsandinoDev) [inline packages/snfoundry/contracts/Scarb.lock:1]: Do not upload this file

File: packages/snfoundry/contracts/Scarb.toml [hunk 1]
```diff
@@ -25,3 +26,6 @@ sort-module-level-items = true
 name = "SEPOLIA_LATEST"
 url = "https://starknet-sepolia.public.blastapi.io/rpc/v0_7"
 block_id.tag = "latest"
+
+# [scripts]
+# test = "snforge test"
```
Comment (jsandinoDev) [inline packages/snfoundry/contracts/Scarb.toml:31]: Remove comments

**Review by jsandinoDev — COMMENTED** [review]
Fix commetns

File: packages/snfoundry/contracts/Scarb.lock
Comment (emarc99) [reply packages/snfoundry/contracts/Scarb.lock:1]: Alright, bet.
